pkg/handler3/handler_test.go

```diff
@@ -82,19 +83,108 @@ func TestRegisterOpenAPIVersionedService(t *testing.T) {
 		respStatus   int
 		urlPath      string
 		respBody     []byte
+		expectedETag string
+		sendETag     bool
```

Comment:
instead of `sendETag`, you could have a `IfNoneMatch` that is configurable (possibly by setting the same value), so that you can test the behavior of the cache at the same time? Though I think you have another test somewhere that does that?